interfaces/kmod/kmod.go

```diff
@@ -18,3 +18,58 @@
  */
 
 package kmod
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+
+	"github.com/snapcore/snapd/dirs"
+	"github.com/snapcore/snapd/osutil"
+	"github.com/snapcore/snapd/snap"
+)
+
+func moduleFilePath(snapName string) string {
+	confFileName := fmt.Sprintf("%s.conf", snap.SecurityTag(snapName))
+	return filepath.Join(dirs.SnapKModModulesDir, confFileName)
+}
+
+// WriteModulesFile creates a .conf file with list of modules for given snap.
+func writeModulesFile(modules [][]byte, snapName string) error {
+	var buffer bytes.Buffer
+	buffer.WriteString("# This file is automatically generated.\n")
+	for _, module := range modules {
+		buffer.Write(module)
+		buffer.WriteByte('\n')
+	}
+
+	content := &osutil.FileState{
+		Content: buffer.Bytes(),
+		Mode:    0644,
+	}
+
+	return osutil.EnsureFileState(moduleFilePath(snapName), content)
+}
+
+// RemoveModulesFile removes modules .conf file for given snap.
+// No error is reported if the file doesn't exist.
+func removeModulesFile(snapName string) error {
```

Comment:
This is also handled by `EnsureDirState`, not sure but I suspect that if you go all the way the module code will be much shorter.